modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/RestMultiSearchTemplateAction.java

```diff
@@ -89,7 +89,8 @@ public static MultiSearchTemplateRequest parseRequest(RestRequest restRequest, b
         RestMultiSearchAction.parseMultiLineRequest(restRequest, multiRequest.indicesOptions(), allowExplicitIndex,
                 (searchRequest, bytes) -> {
                     if (searchRequest.types().length > 0) {
-                        deprecationLogger.deprecated(RestMultiSearchAction.TYPES_DEPRECATION_MESSAGE);
+                        deprecationLogger.deprecatedAndMaybeLog("msearch_template_with_types",
+                            RestMultiSearchAction.TYPES_DEPRECATION_MESSAGE);
```

Comment:
Do we want to use `RestMultiSearchTemplateAction` here and the same with logger on line 44?

Reply:
Thanks for pointing this out @mayya-sharipova. If it's okay with you, I'll plan to fix this in another PR as this one is already reasonably large.

Reply:
An update: I've fixed this in #36344.